lib/charms/archive_auth_mirror/utils.py

```diff
@@ -53,8 +58,23 @@ def install_resources(base_dir=None):
     for name in ('bin', 'reprepro-conf', 'static', 'gnupghome'):
         paths[name].mkdir(parents=True, exist_ok=True)
 
-    # create an empty basic-auth password file, only readable by root
-    paths['basic-auth'].touch(mode=0o400)
+    # make directories accessible by the web server
+    for name in ('base', 'static'):
+        _chgrp(paths[name], 'www-data')
+
+    # create an empty basic-auth password file, readable by the web server but
+    # only writable by root
+    paths['basic-auth'].touch(mode=0o640)
+    _chgrp(str(paths['basic-auth']), 'www-data')
+
     # copy scripts
-    shutil.copy('resources/mirror-archive', str(paths['bin']))
-    shutil.copy('resources/manage-user', str(paths['bin']))
+    for resource in ('mirror-archive', 'manage-user'):
+        resource_path = os.path.join('resources', resource)
+        shutil.copy(resource_path, str(paths['bin']))
+
+
+def _chgrp(path, group):
+    '''Change group ownership to a file/directory.'''
+    path = str(path)
+    gid = grp.getgrnam(group).gr_gid
+    os.chown(path, -1, gid)
```

Comment:
By using [shutil.chown](https://docs.python.org/3/library/shutil.html#shutil.chown) we can avoid the need to lookup the gid